libraries/shared/watcher/storage_watcher.go

```diff
@@ -63,6 +71,7 @@ func NewStorageWatcher(db *postgres.DB, backFromHeadOfChain int64, statusWriter
 		StorageDiffRepository:     storageDiffRepository,
 		DiffBlocksFromHeadOfChain: backFromHeadOfChain,
 		StatusWriter:              statusWriter,
+		DiffStatus:                diffStatusToWatch,
```

Comment:
not sure if this would make things less clear but I wonder if we could directly pass the function that should be called to get diffs, instead of a flag

Reply:
🤔 I really like this idea. I may hold off, and do it as part of a new PR/story. I think it will require a bit of reworking since the `GetNewDiffs` and `GetUnrecognizedDiffs` are methods on the `diffRepository` which the execute command doesn't know about. Open to thoughts on how to get around this without a bunch of reworking how the repository works!